HealthcareSystem/Backend/Model/Users/City.cs

```diff
@@ -24,15 +24,36 @@ public class City
         public virtual ICollection<User> Users { get; set; }
 
         public City() { }
-        public City(string name,int zipCode)
+        public City(int zipCode,string name,Country country)
         {
-            Name = name;
             ZipCode = zipCode;
+            Name = name;
+            if (country != null)
+            {
+                Country = new Country(country);
+                CountryId = country.Id;
+            }
+            else
+            {
+                Country = new Country();
+                CountryId = 0;
+            }
+            
         }
         public City(City city)
         {
-            Name = city.Name;
             ZipCode = city.ZipCode;
+            Name = city.Name;
+            if (city.Country != null)
+            {
+                Country = new Country(city.Country);
+                CountryId = city.Country.Id;
+            }
+            else
+            {
+                Country = new Country();
+                CountryId = 0;
+            }
         }
     }
```

Comment:
This is good, just one suggestion to stop DRY, try to extract method something like 'InitCountry' or some better description for it. And then use it in constructors.

```sh
        public City(int zipCode,string name,Country country)
        {
            ZipCode = zipCode;
            Name = name;
            InitCountry(country);
        }

        public City(City city)
        {
            ZipCode = city.ZipCode;
            Name = city.Name;
            InitCountry(city.Country);

        }

        private void InitCountry(Country country)
        {
            if (country != null)
            {
                Country = new Country(country);
                CountryId = country.Id;
            }
            else
            {
                Country = new Country();
                CountryId = 0;
            }
        }

```